charts/plugin-health-scoring/templates/deployment.yaml

```diff
@@ -63,11 +63,15 @@ spec:
                 secretKeyRef:
                   name: {{ include "plugin-health-scoring.fullname" . }}
                   key: databasePassword
-            - name: GITHUB_OAUTH
+            - name: GITHUB_APP_ID
+              value: {{ .Values.config.github.appId | squote }}
+            - name: GITHUB_APP_PRIVATE_KEY_PATH
               valueFrom:
                 secretKeyRef:
                   name: {{ include "plugin-health-scoring.fullname" . }}
-                  key: githubOauthToken
+                  key: githubPrivateKey
```

Comment:
You might want to have a look at https://kubernetes.io/docs/concepts/configuration/secret/#projection-of-secret-keys-to-specific-paths

- The (multiline) key value will be passed through a helm value (see my suggestion to provide a dummy multiline key as default value)
- The value will be base64 encoded (easier to manage multi line strings) and stores in the Kubernetes Secret as a sub key of `data:`
- The deployment will specify the secret (restricted to the sub key) as a volume
- The container of the pod described by the deployment will mount the secret in a predefined path that will be fed to the environment variable here => this "predefined" path can be a helm value (to set in values.yaml) or a static value (to define in the _helpers.tpl file)